R/zchunk_L251.en_ssp_nonco2.R

```diff
@@ -35,106 +40,215 @@ module_emissions_L251.en_ssp_nonco2_DISABLED <- function(command, ...) {
+             year = 1975)->
+      L251.ssp2_ef_vin
+
+    L251.ssp34_ef %>%
+      # Isolate the "electricity" supply sector.
+      filter(supplysector == "electricity") %>%
+      # Create 2 new columns for future emission factors.
+      # Future emission coefficient year is based on pre-existing "year" column.
+      # Future emission coefficient name is a descriptor, and is constant.
+      mutate(future.emiss.coeff.year = year,
+             future.emiss.coeff.name = "SSP_GAINS",
+             # Previous year column that only includes the electricity supply sector is now a constant.
+             year = 1975)->
+      L251.ssp34_ef_vin
+
+    # This section renames SO2 variables so that it has regional SO2 emission species.
+    L251.ctrl.delete <- rename_SO2(L251.ctrl.delete, A_regions, FALSE)
+    L251.ssp15_ef <- rename_SO2(L251.ssp15_ef, A_regions, FALSE)
+    L251.ssp2_ef <- rename_SO2(L251.ssp2_ef, A_regions, FALSE)
+    L251.ssp34_ef <- rename_SO2(L251.ssp34_ef, A_regions, FALSE)
+    L251.ssp15_ef_vin <- rename_SO2(L251.ssp15_ef_vin, A_regions, FALSE)
+    L251.ssp2_ef_vin <- rename_SO2(L251.ssp2_ef_vin, A_regions, FALSE)
+    L251.ssp34_ef_vin <- rename_SO2(L251.ssp34_ef_vin, A_regions, FALSE)
+
+    # This section deletes GDP control functions that exist for the 1975 control period.
+    L251.ctrl.delete %>%
+      semi_join(L201.nonghg_steepness,
+                by = c("region", "supplysector", "subsector", "stub.technology", "Non.CO2")) ->
+      L251.ctrl.delete
+
+    # ===================================================
     # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # No flags are necessary because old data is in 'long' format.
+
+    L251.ctrl.delete %>%
+      add_title("GDP control of regional non-CO2 emissions.") %>%
+      add_units("unitless") %>%
+      add_comments("Make year 2010 the control period in creation of GDP control data.") %>%
+      add_comments("Then, created a new column for data that has regional non-CO2 emission species.") %>%
+      add_comments("Then, delete GDP control functions that exist.") %>%
       add_legacy_name("L251.ctrl.delete") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("temp-data-inject/L161.SSP2_EF", "temp-data-inject/L201.nonghg_steepness", "emissions/A_regions") ->
```

Comment:
Is it that each precursor has its own line? rather than including all precursors in 1 line?